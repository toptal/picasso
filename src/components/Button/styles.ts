import { Theme, createStyles } from '@material-ui/core/styles'

import { lighten, darken, alpha } from '../styles'

const ICON_SPACING = '0.4em'

const primary = (mainColor: string, secondaryColor: string) => ({
  border: 'none',
  color: secondaryColor,
  backgroundColor: mainColor,

  '&:hover, &$hovered': {
    backgroundColor: darken(mainColor, 0.2)
  },

  '&:active, &$active': {
    backgroundColor: darken(mainColor, 0.2)
  }
})

const secondary = (mainColor: string, secondaryColor: string) => ({
  color: mainColor,
  backgroundColor: secondaryColor,

  '&:hover, &$hovered': {
    backgroundColor: lighten(mainColor, 0.84),
    borderColor: mainColor
  },

  '&:active, &$active': {
    backgroundColor: lighten(mainColor, 0.84),
    borderColor: mainColor
  }
})

export default ({ palette, spacing, transitions, typography }: Theme) =>
  createStyles({
    root: {
      position: 'relative',
      textTransform: 'none',
      borderRadius: spacing.borderRadius,
      border: `solid ${spacing.borderWidth} ${palette.grey[100]}`,
      fontSize: '1rem',
      transition: `all ${transitions.duration.short}ms ${
        transitions.easing.easeOut
      }`,
      transitionProperty: 'border, color, background',

      '&:focus, &$focused': {
        textDecoration: 'underline',

        '&:active, &$active, &:hover': {
          textDecoration: 'none'
        }
      },

      '&+&': {
        marginLeft: '0.5em'
      }
    },
    content: {
      lineHeight: '1.5em',
      fontWeight: typography.fontWeights.semibold
    },
    loader: {
      position: 'absolute',
      left: '50%',
      top: '50%',
      transform: 'translate(-50%, -50%)'
    },

    // sizes
    small: {
      height: '1.5em',
      padding: '0 0.75em',

      '& $content': {
        fontSize: typography.buttons.fontSizeSmall
      },

      '&$circular': {
        width: '1.5em'
      }
    },
    medium: {
      height: '2.25em',
      padding: '0 1em',

      '& $content': {
        fontSize: typography.buttons.fontSizeMedium
      },

      '&$circular': {
        width: '2.25em'
      }
    },
    large: {
      height: '3em',
      padding: '0 3.625em',

      '& $content': {
        fontSize: typography.buttons.fontSizeLarge,
        fontWeight: typography.fontWeights.semibold
      },

      '&$circular': {
        width: '3em'
      }
    },

    // variants
    primaryBlue: primary(palette.primary.main, palette.common.white),
    secondaryBlue: secondary(palette.primary.main, palette.common.white),
    primaryRed: primary(palette.error.main, palette.common.white),
    secondaryRed: secondary(palette.error.main, palette.common.white),
    primaryGreen: primary(palette.success.main, palette.common.white),
    secondaryWhite: {
      ...secondary(palette.common.white, palette.common.white),
      backgroundColor: 'transparent',
      border: `solid ${spacing.borderWidth} rgba(255, 255, 255, 0.32)`,

      '&:hover, &$hovered': {
        backgroundColor: alpha(palette.common.white, 0.16),
        borderColor: palette.common.white
      },

      '&:active, &$active': {
        backgroundColor: alpha(palette.common.white, 0.16),
        borderColor: palette.common.white
      }
    },
    flat: {
      ...secondary(palette.common.black, palette.common.white),
      border: 'none'
    },
    primaryDisabled: primary(palette.grey[100], palette.common.white),
    secondaryDisabled: secondary(palette.grey[100], palette.common.white),
    flatDisabled: {
      ...secondary(palette.grey[100], palette.common.white),
      border: 'none'
    },

    // Other props
    fullWidth: {
      width: '100%'
    },
    hovered: {},
    focused: {},
    active: {},
    circular: {
      borderRadius: '50%',
      padding: 0
    },

    // Child elements
    icon: {
      fontSize: '1.2em !important'
    },
    iconLeft: {
      marginRight: ICON_SPACING
    },
    iconRight: {
      marginLeft: ICON_SPACING
    },
    hidden: {
      opacity: 0
    }
  })
