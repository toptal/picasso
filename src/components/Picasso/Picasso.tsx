import {
  createMuiTheme,
  MuiThemeProvider,
  withStyles
} from '@material-ui/core/styles'
import React, { FunctionComponent, ReactNode } from 'react'

import CssBaseline from '../CssBaseline'
import {
  palette,
  layout,
  transitions,
  typography,
  spacing,
  breakpoints,
  screens,
  shadows
} from './config'
import FontsLoader from './FontsLoader'
import Provider from './PicassoProvider'
import globalStyles from './styles'
import { JssProps } from './types'

const picasso = {
  palette,
  layout,
  transitions,
  spacing,
  breakpoints,
  screens,
  shadows,
  typography,
  props: {
    MuiButtonBase: {
      disableRipple: true
    },
    MuiList: {
      disablePadding: true
    },
    MuiPaper: {
      square: true
    },
    MuiOutlinedInput: {
      notched: false
    }
  }
}

const PicassoProvider = new Provider(createMuiTheme(picasso))

interface PicassoGlobalStylesProviderProps extends JssProps {
  children?: ReactNode
}

const PicassoGlobalStylesProvider = withStyles(globalStyles, {
  name: 'Picasso'
})((props: PicassoGlobalStylesProviderProps) => {
  const { classes, children } = props

  return <div className={classes.root}>{children}</div>
})

interface PicassoProps {
  children?: ReactNode
  /** Whether to load fonts file to the page */
  loadFonts?: boolean
  /** Whether to apply Picasso CSS reset */
  reset?: boolean
}

const Picasso: FunctionComponent<PicassoProps> = ({
  loadFonts,
  reset,
  children
}) => (
  <MuiThemeProvider theme={PicassoProvider.theme}>
    {loadFonts && <FontsLoader />}
    {reset && <CssBaseline />}
    <PicassoGlobalStylesProvider>{children}</PicassoGlobalStylesProvider>
  </MuiThemeProvider>
)

Picasso.defaultProps = {
  loadFonts: true,
  reset: true
}

export { PicassoProvider }
export default Picasso
