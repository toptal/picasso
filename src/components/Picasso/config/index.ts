import './theme'
export { default as palette } from './palette'
export { default as transitions } from './transitions'
export { default as typography } from './typography'
export { default as sizes } from './sizes'
export {
  default as breakpoints,
  screens,
  useScreenSize,
  isScreenSize,
  useBreakpoint
} from './breakpoints'
export { default as layout } from './layout'
export { default as shadows } from './shadows'
