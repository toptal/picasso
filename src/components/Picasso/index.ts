export {
  default,
  PicassoProvider,
  usePicassoRoot,
  usePageHeader
} from './Picasso'

export {
  BaseProps,
  JssProps,
  StandardProps,
  SizeType,
  OmitInternalProps,
  SpacingType,
  spacingToEm,
  PicassoComponent,
  PicassoComponentWithRef,
  SpacingEnum,
  ButtonOrAnchorProps,
  CompoundedComponentWithRef,
  OverridableComponent,
  ColorType
} from './types'

export { useScreenSize, isScreenSize, useBreakpoint } from './config'
