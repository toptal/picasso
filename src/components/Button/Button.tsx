import React, { ReactNode, ReactElement, MouseEvent, forwardRef } from 'react'
import cx from 'classnames'
import { withStyles } from '@material-ui/core/styles'
import ButtonBase from '@material-ui/core/ButtonBase'

import Loader from '../Loader'
import Container from '../Container'
import Group from '../ButtonGroup'
import kebabToCamelCase from '../utils/kebab-to-camel-case'
import {
  StandardProps,
  PicassoComponentWithRef,
  SizeType,
  ButtonOrAnchorProps,
  CompoundedComponentWithRef
} from '../Picasso'
import styles from './styles'

type VariantType =
  | 'primary-blue'
  | 'secondary-blue'
  | 'primary-red'
  | 'secondary-red'
  | 'primary-green'
  | 'flat'
  | 'flat-white'
  | 'secondary-white'

type IconPositionType = 'left' | 'right'

export interface Props extends StandardProps, ButtonOrAnchorProps {
  /** Show button in the active state (left mouse button down) */
  active?: boolean
  /** Disables button */
  disabled?: boolean
  /** Content of Button component */
  children?: ReactNode
  // TODO: should it be exposed?
  focused?: boolean
  /** Take the full width of a container */
  fullWidth?: boolean
  /** Set hovered style for the button */
  hovered?: boolean
  /** Add an `<Icon />` along Button's children */
  icon?: ReactElement
  /** Icon can be positioned on the left or right */
  iconPosition?: IconPositionType
  /** A button can show a loading indicator */
  loading?: boolean
  /** Callback invoked when component is clicked */
  onClick?: (event: MouseEvent<HTMLButtonElement & HTMLAnchorElement>) => void
  /** A button can have different sizes */
  size?: SizeType<'small' | 'medium' | 'large'>
  /** The variant to use */
  variant?: VariantType
  /** HTML Value of Button component */
  value?: string | number
  /** Circular style of Button component */
  circular?: boolean
  /** HTML title of Button component */
  title?: string
  /** HTML type of Button component **/
  type?: 'button' | 'reset' | 'submit'
}

interface StaticProps {
  Group: typeof Group
}

const getVariantType = (variant: VariantType) => {
  const [type] = variant!.split('-')

  return type
}

export const Button = forwardRef<HTMLButtonElement, Props>(function Button(
  {
    icon,
    iconPosition,
    loading,
    children,
    classes,
    className,
    style,
    fullWidth,
    variant,
    size,
    focused,
    hovered,
    disabled,
    active,
    onClick,
    circular,
    title,
    value,
    type,
    ...rest
  },
  ref
) {
  const {
    icon: iconClass,
    iconLeft: iconLeftClass,
    iconRight: iconRightClass,
    iconSmall: iconSmallClass,
    root: rootClass,
    hidden: hiddenClass,
    loader: loaderClass,
    content: contentClass
  } = classes

  let finalChildren = [children]

  if (icon) {
    const iconComponent = React.cloneElement(icon, {
      className: cx(iconClass, icon.props.className, {
        [iconLeftClass]: children && iconPosition === 'left',
        [iconRightClass]: children && iconPosition === 'right',
        [iconSmallClass]: size === 'small'
      }),
      key: 'button-icon',
      base: size === 'small' ? '12' : undefined
    })

    if (iconPosition === 'left') {
      finalChildren.unshift(iconComponent)
    } else {
      finalChildren.push(iconComponent)
    }
  }

  const variantType = getVariantType(variant!)
  const variantClassName = disabled
    ? classes[`${variantType}Disabled`]
    : classes[kebabToCamelCase(variant!)]
  const sizeClassName = classes[size!]

  const rootClassName = cx(
    {
      [classes.fullWidth]: fullWidth,
      [classes.active]: active,
      [classes.focused]: focused,
      [classes.hovered]: hovered,
      [classes.circular]: circular
    },
    sizeClassName,
    variantClassName,
    rootClass
  )

  return (
    <ButtonBase
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...rest}
      ref={ref}
      classes={{
        root: rootClassName
      }}
      onClick={onClick}
      className={className}
      style={style}
      disabled={disabled}
      title={title}
      value={value}
      type={type}
    >
      <Container
        as='span'
        inline
        flex
        direction='row'
        alignItems='center'
        className={cx({ [hiddenClass]: loading }, contentClass)}
      >
        {finalChildren}
      </Container>

      {loading && (
        <Loader variant='inherit' className={loaderClass} inline size='small' />
      )}
    </ButtonBase>
  )
}) as CompoundedComponentWithRef<Props, HTMLButtonElement, StaticProps>

Button.defaultProps = {
  active: false,
  children: null,
  circular: false,
  disabled: false,
  focused: false,
  fullWidth: false,
  hovered: false,
  iconPosition: 'left',
  loading: false,
  onClick: () => {},
  size: 'medium',
  type: 'button',
  variant: 'primary-blue'
}

Button.displayName = 'Button'

Button.Group = Group

export default withStyles(styles)(Button) as PicassoComponentWithRef<
  Props,
  HTMLButtonElement,
  StaticProps
>
