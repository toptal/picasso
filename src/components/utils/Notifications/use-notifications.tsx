import React, { ReactNode, ReactElement } from 'react'
import cx from 'classnames'
import { useSnackbar, OptionsObject } from 'notistack'
import { withStyles } from '@material-ui/core/styles'
import { SnackbarOrigin } from '@material-ui/core/Snackbar'

import Notification, { VariantType } from '../../Notification'
import { Classes } from '../../styles/types'
import styles from './styles'

const defaultPosition: SnackbarOrigin = {
  vertical: 'top',
  horizontal: 'right'
}

interface Props {
  key: string
  content: ReactNode
  icon?: ReactElement
  classes: Classes
  onClose?: () => void
  variant?: VariantType
}

const StyledNotification = withStyles(styles)(
  ({ content, icon, key, onClose, variant = 'white', classes }: Props) => (
    <Notification
      variant={variant}
      elevated
      icon={icon}
      key={key}
      onClose={onClose}
      className={cx({
        [classes.generalNotification]: variant === 'white',
        [classes.formNotification]: variant !== 'white'
      })}
    >
      {content}
    </Notification>
  )
)

export const useNotifications = () => {
  const { enqueueSnackbar, closeSnackbar } = useSnackbar()

  const getNotification = (variant?: VariantType) => (
    content: ReactNode,
    icon?: ReactElement,
    options?: OptionsObject
  ) => {
    const closeNotification = () => {
      if (!notificationId) {
        return
      }

      closeSnackbar(notificationId)
    }
    const notificationId = enqueueSnackbar('', {
      anchorOrigin: defaultPosition,
      // eslint-disable-next-line react/display-name
      children: (key: string) => (
        <StyledNotification
          content={content}
          icon={icon}
          key={key}
          variant={variant}
          onClose={closeNotification}
        />
      ),
      ...options
    })

    return notificationId
  }

  const showCustomNotification = (
    Content: ReactElement,
    position?: SnackbarOrigin,
    options?: OptionsObject
  ) =>
    enqueueSnackbar('', {
      anchorOrigin: position || defaultPosition,
      // eslint-disable-next-line react/display-name
      children: (key: string) => React.cloneElement(Content, { key }),
      ...options
    })

  return {
    showError: getNotification('red'),
    showInfo: getNotification(),
    showSuccess: getNotification('green'),
    showCustomNotification: showCustomNotification,
    closeNotification: closeSnackbar
  }
}
