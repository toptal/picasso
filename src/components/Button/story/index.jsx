import PicassoBook from '~/.storybook/components/PicassoBook'

import buttonGroupStory from '@components/ButtonGroup/story'

import { Button } from '../Button'

const page = PicassoBook.createPage(
  'Button',
  'A Button indicates a possible user action.'
)

page
  .createTabChapter('Props')
  .addComponentDocs({
    component: Button,
    additionalDocs: {
      icon: {
        type: 'ReactElement'
      }
    },
    name: 'Button'
  })
  .addComponentDocs(buttonGroupStory.componentDocs)

page
  .createChapter()
  .addExample('Button/story/Basic.example.jsx', 'Basic')
  .addExample('Button/story/Variants.example.jsx', 'Variants')
  .addExample('Button/story/States.example.jsx', 'States')
  .addExample('Button/story/Disabled.example.jsx', {
    title: 'Disabled',
    description: 'The button shows that currently unable to be interacted with'
  })
  .addExample('Button/story/Sizes.example.jsx', 'Sizes')
  .addExample('Button/story/FullWidth.example.jsx', 'Full width')
  .addExample('Button/story/IconButtons.example.jsx', 'Button with Icon')
  .addExample(
    'Button/story/CircularIconButton.example.jsx',
    'Circular Style Button'
  )
  .addExample(
    'Button/story/IconButtonsWithText.example.jsx',
    'Button with text and Icon'
  )
  .addExample('Button/story/Loading.example.jsx', 'Button with loading state')

page.connect(buttonGroupStory.chapter)
