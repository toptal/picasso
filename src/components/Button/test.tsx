import React, { ReactNode } from 'react'
import { render, fireEvent, cleanup, RenderResult } from 'react-testing-library'

import Picasso, { OmitInternalProps } from '../Picasso'
import Button, { Props } from './Button'

const renderButton = (children: ReactNode, props: OmitInternalProps<Props>) => {
  const { disabled, onClick } = props

  return render(
    <Picasso loadFonts={false}>
      <Button disabled={disabled} onClick={onClick}>
        {children}
      </Button>
    </Picasso>
  )
}

afterEach(cleanup)

test('onClick callback should be fired after clicking the button', () => {
  const onClick = jest.fn()
  const { getByText } = renderButton('Click me!', { onClick })

  fireEvent.click(getByText('Click me!'))

  expect(onClick).toHaveBeenCalled()
})

describe('disabled button', () => {
  let onClick: () => void
  let api: RenderResult

  beforeEach(() => {
    onClick = jest.fn()

    api = renderButton('Click me!', {
      onClick,
      disabled: true
    })
  })
  test('renders disabled version', () => {
    const { container } = api

    expect(container).toMatchSnapshot()
  })

  test('disables button events', () => {
    const { getByText } = api

    fireEvent.click(getByText('Click me!'))

    expect(onClick).not.toHaveBeenCalled()
  })
})
