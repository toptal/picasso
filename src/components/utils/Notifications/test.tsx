import React, { Fragment } from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'

import Picasso from '../../Picasso'
import { useNotifications } from './use-notifications'
import Button from '../../Button'

const TestComponent = () => {
  const { showError, showInfo, showSuccess } = useNotifications()

  return (
    <Fragment>
      <Button onClick={() => showError('Error message')}>Show Error</Button>
      <Button onClick={() => showInfo('Info message')}>Show Info</Button>
      <Button onClick={() => showSuccess('Success message')}>
        Show Success
      </Button>
    </Fragment>
  )
}

const renderNotification = () => {
  return render(
    <Picasso loadFonts={false}>
      <TestComponent />
    </Picasso>
  )
}

afterEach(cleanup)

describe('useNotifications', () => {
  test('error notification render', () => {
    const { getByText, container } = renderNotification()
    const showErrorButton = getByText('Show Error')

    fireEvent.click(showErrorButton)

    expect(container).toMatchSnapshot()
  })

  test('info notification render', () => {
    const { getByText, container } = renderNotification()
    const showInfoButton = getByText('Show Info')

    fireEvent.click(showInfoButton)

    expect(container).toMatchSnapshot()
  })

  test('success notification render', () => {
    const { getByText, container } = renderNotification()
    const showSuccessButton = getByText('Show Success')

    fireEvent.click(showSuccessButton)

    expect(container).toMatchSnapshot()
  })
})
