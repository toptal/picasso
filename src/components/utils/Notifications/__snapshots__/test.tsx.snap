// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useNotifications error notification render 1`] = `
<div>
  <div
    class="Picasso-root"
  >
    <button
      class="MuiButtonBase-root Button-medium Button-primaryBlue Button-root"
      tabindex="0"
      type="button"
    >
      <span
        class="Component-flex Component-inline Button-content"
        style="align-items: center;"
      >
        Show Error
      </span>
    </button>
    <button
      class="MuiButtonBase-root Button-medium Button-primaryBlue Button-root"
      tabindex="0"
      type="button"
    >
      <span
        class="Component-flex Component-inline Button-content"
        style="align-items: center;"
      >
        Show Info
      </span>
    </button>
    <button
      class="MuiButtonBase-root Button-medium Button-primaryBlue Button-root"
      tabindex="0"
      type="button"
    >
      <span
        class="Component-flex Component-inline Button-content"
        style="align-items: center;"
      >
        Show Success
      </span>
    </button>
    <div
      class="MuiSnackbar-root SnackbarItem-root MuiSnackbar-anchorOriginTopRight SnackbarItem-anchorOriginTopRight"
      height="0"
      style="top: 20px; transition: all 200ms; transition-delay: 150ms;"
    >
      <div
        class="MuiTypography-root MuiTypography-body2 MuiPaper-root MuiPaper-elevation6 MuiSnackbarContent-root Notification-notificationRed Notification-notificationShadow Notification-notification Component-formNotification"
        role="alertdialog"
        style="webkit-transform: translate(0, 0); transform: translate(0, 0); webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiSnackbarContent-message"
        >
          <div
            class="Component-flex Notification-iconWrapper"
            style="align-items: center;"
          >
            <svg
              class="SvgExclamation16-root"
              color="#d42551"
              style="min-width: 16px; min-height: 16px;"
              viewBox="0 0 16 16"
            >
              <defs>
                <path
                  d="M7.5 15a7.5 7.5 0 1 1 0-15 7.5 7.5 0 0 1 0 15zm0-1a6.5 6.5 0 1 0 0-13 6.5 6.5 0 0 0 0 13zM7 3h1v6H7V3zm0 8h1v1H7v-1z"
                  id="exclamation16_svg__a"
                />
              </defs>
              <g
                fill-rule="evenodd"
              >
                <mask
                  id="exclamation16_svg__b"
                >
                  <use
                    xlink:href="#exclamation16_svg__a"
                  />
                </mask>
                <use
                  xlink:href="#exclamation16_svg__a"
                />
                <g
                  mask="url(#exclamation16_svg__b)"
                >
                  <path
                    d="M0 0h16v16H0z"
                  />
                </g>
              </g>
            </svg>
          </div>
          <div
            class="MuiTypography-root Typography-bodyMedium MuiTypography-body1 Notification-content Notification-contentCloseButton"
          >
            Error message
          </div>
          <button
            class="MuiButtonBase-root Button-circular Button-medium Button-primaryBlue Button-root Notification-close"
            tabindex="0"
            title="Close Notification"
            type="button"
          >
            <span
              class="Component-flex Component-inline Button-content"
              style="align-items: center;"
            >
              <svg
                class="SvgCloseMinor16-root Button-icon Notification-closeIcon"
                style="min-width: 16px; min-height: 16px;"
                viewBox="0 0 16 16"
              >
                <defs>
                  <path
                    d="M8.707 8l3.5 3.5-.707.707-3.5-3.5-3.5 3.5-.707-.707 3.5-3.5-3.5-3.5.707-.707 3.5 3.5 3.5-3.5.707.707-3.5 3.5z"
                    id="closeMinor16_svg__a"
                  />
                </defs>
                <g
                  fill-rule="evenodd"
                >
                  <mask
                    id="closeMinor16_svg__b"
                  >
                    <use
                      xlink:href="#closeMinor16_svg__a"
                    />
                  </mask>
                  <use
                    fill-rule="nonzero"
                    xlink:href="#closeMinor16_svg__a"
                  />
                  <g
                    mask="url(#closeMinor16_svg__b)"
                  >
                    <path
                      d="M0 0h16v16H0z"
                    />
                  </g>
                </g>
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`useNotifications info notification render 1`] = `
<div>
  <div
    class="Picasso-root"
  >
    <button
      class="MuiButtonBase-root Button-medium Button-primaryBlue Button-root"
      tabindex="0"
      type="button"
    >
      <span
        class="Component-flex Component-inline Button-content"
        style="align-items: center;"
      >
        Show Error
      </span>
    </button>
    <button
      class="MuiButtonBase-root Button-medium Button-primaryBlue Button-root"
      tabindex="0"
      type="button"
    >
      <span
        class="Component-flex Component-inline Button-content"
        style="align-items: center;"
      >
        Show Info
      </span>
    </button>
    <button
      class="MuiButtonBase-root Button-medium Button-primaryBlue Button-root"
      tabindex="0"
      type="button"
    >
      <span
        class="Component-flex Component-inline Button-content"
        style="align-items: center;"
      >
        Show Success
      </span>
    </button>
    <div
      class="MuiSnackbar-root SnackbarItem-root MuiSnackbar-anchorOriginTopRight SnackbarItem-anchorOriginTopRight"
      height="0"
      style="top: 20px; transition: all 200ms; transition-delay: 150ms;"
    >
      <div
        class="MuiTypography-root MuiTypography-body2 MuiPaper-root MuiPaper-elevation6 MuiSnackbarContent-root Notification-notificationWhite Notification-notificationShadow Notification-notification Component-generalNotification"
        role="alertdialog"
        style="webkit-transform: translate(0, 0); transform: translate(0, 0); webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiSnackbarContent-message"
        >
          <div
            class="Component-flex Notification-iconWrapper"
            style="align-items: center;"
          >
            <svg
              class="SvgInfo16-root"
              color="#c4c6ca"
              style="min-width: 16px; min-height: 16px;"
              viewBox="0 0 16 16"
            >
              <defs>
                <path
                  d="M7.5 15a7.5 7.5 0 1 1 0-15 7.5 7.5 0 0 1 0 15zm0-1a6.5 6.5 0 1 0 0-13 6.5 6.5 0 0 0 0 13zM7 6h1v6H7V6zm0-3h1v1H7V3z"
                  id="info16_svg__a"
                />
              </defs>
              <g
                fill-rule="evenodd"
              >
                <mask
                  id="info16_svg__b"
                >
                  <use
                    xlink:href="#info16_svg__a"
                  />
                </mask>
                <use
                  xlink:href="#info16_svg__a"
                />
                <g
                  mask="url(#info16_svg__b)"
                >
                  <path
                    d="M0 0h16v16H0z"
                  />
                </g>
              </g>
            </svg>
          </div>
          <div
            class="MuiTypography-root Typography-bodyMedium MuiTypography-body1 Notification-content Notification-contentCloseButton"
          >
            Info message
          </div>
          <button
            class="MuiButtonBase-root Button-circular Button-medium Button-primaryBlue Button-root Notification-close"
            tabindex="0"
            title="Close Notification"
            type="button"
          >
            <span
              class="Component-flex Component-inline Button-content"
              style="align-items: center;"
            >
              <svg
                class="SvgCloseMinor16-root Button-icon Notification-closeIcon"
                style="min-width: 16px; min-height: 16px;"
                viewBox="0 0 16 16"
              >
                <defs>
                  <path
                    d="M8.707 8l3.5 3.5-.707.707-3.5-3.5-3.5 3.5-.707-.707 3.5-3.5-3.5-3.5.707-.707 3.5 3.5 3.5-3.5.707.707-3.5 3.5z"
                    id="closeMinor16_svg__a"
                  />
                </defs>
                <g
                  fill-rule="evenodd"
                >
                  <mask
                    id="closeMinor16_svg__b"
                  >
                    <use
                      xlink:href="#closeMinor16_svg__a"
                    />
                  </mask>
                  <use
                    fill-rule="nonzero"
                    xlink:href="#closeMinor16_svg__a"
                  />
                  <g
                    mask="url(#closeMinor16_svg__b)"
                  >
                    <path
                      d="M0 0h16v16H0z"
                    />
                  </g>
                </g>
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`useNotifications success notification render 1`] = `
<div>
  <div
    class="Picasso-root"
  >
    <button
      class="MuiButtonBase-root Button-medium Button-primaryBlue Button-root"
      tabindex="0"
      type="button"
    >
      <span
        class="Component-flex Component-inline Button-content"
        style="align-items: center;"
      >
        Show Error
      </span>
    </button>
    <button
      class="MuiButtonBase-root Button-medium Button-primaryBlue Button-root"
      tabindex="0"
      type="button"
    >
      <span
        class="Component-flex Component-inline Button-content"
        style="align-items: center;"
      >
        Show Info
      </span>
    </button>
    <button
      class="MuiButtonBase-root Button-medium Button-primaryBlue Button-root"
      tabindex="0"
      type="button"
    >
      <span
        class="Component-flex Component-inline Button-content"
        style="align-items: center;"
      >
        Show Success
      </span>
    </button>
    <div
      class="MuiSnackbar-root SnackbarItem-root MuiSnackbar-anchorOriginTopRight SnackbarItem-anchorOriginTopRight"
      height="0"
      style="top: 20px; transition: all 200ms; transition-delay: 150ms;"
    >
      <div
        class="MuiTypography-root MuiTypography-body2 MuiPaper-root MuiPaper-elevation6 MuiSnackbarContent-root Notification-notificationGreen Notification-notificationShadow Notification-notification Component-formNotification"
        role="alertdialog"
        style="webkit-transform: translate(0, 0); transform: translate(0, 0); webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiSnackbarContent-message"
        >
          <div
            class="Component-flex Notification-iconWrapper"
            style="align-items: center;"
          >
            <svg
              class="SvgCheckMinor16-root"
              color="#00cc83"
              style="min-width: 16px; min-height: 16px;"
              viewBox="0 0 16 16"
            >
              <defs>
                <path
                  d="M6.5 9.793l4-4 .707.707-4 4-.707.707L3.793 8.5l.707-.707 2 2z"
                  id="checkMinor16_svg__a"
                />
              </defs>
              <g
                fill-rule="evenodd"
              >
                <mask
                  id="checkMinor16_svg__b"
                >
                  <use
                    xlink:href="#checkMinor16_svg__a"
                  />
                </mask>
                <use
                  fill-rule="nonzero"
                  xlink:href="#checkMinor16_svg__a"
                />
                <g
                  mask="url(#checkMinor16_svg__b)"
                >
                  <path
                    d="M0 0h16v16H0z"
                  />
                </g>
              </g>
            </svg>
          </div>
          <div
            class="MuiTypography-root Typography-bodyMedium MuiTypography-body1 Notification-content Notification-contentCloseButton"
          >
            Success message
          </div>
          <button
            class="MuiButtonBase-root Button-circular Button-medium Button-primaryBlue Button-root Notification-close"
            tabindex="0"
            title="Close Notification"
            type="button"
          >
            <span
              class="Component-flex Component-inline Button-content"
              style="align-items: center;"
            >
              <svg
                class="SvgCloseMinor16-root Button-icon Notification-closeIcon"
                style="min-width: 16px; min-height: 16px;"
                viewBox="0 0 16 16"
              >
                <defs>
                  <path
                    d="M8.707 8l3.5 3.5-.707.707-3.5-3.5-3.5 3.5-.707-.707 3.5-3.5-3.5-3.5.707-.707 3.5 3.5 3.5-3.5.707.707-3.5 3.5z"
                    id="closeMinor16_svg__a"
                  />
                </defs>
                <g
                  fill-rule="evenodd"
                >
                  <mask
                    id="closeMinor16_svg__b"
                  >
                    <use
                      xlink:href="#closeMinor16_svg__a"
                    />
                  </mask>
                  <use
                    fill-rule="nonzero"
                    xlink:href="#closeMinor16_svg__a"
                  />
                  <g
                    mask="url(#closeMinor16_svg__b)"
                  >
                    <path
                      d="M0 0h16v16H0z"
                    />
                  </g>
                </g>
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
